qiskit_experiments/calibration/calibration_definitions.py

```diff
@@ -0,0 +1,382 @@
+    A class to manage schedules with calibrated parameter values.
+    Schedules are stored in a dict and are intended to be fully parameterized,
+    including the index of the channels. The parameters are therefore stored in
+    the schedules. The names of the parameters must be unique. The calibrated
+    values of the parameters are stored in a dictionary.
+    """
+
+    def __init__(self, backend):
+        """
+        Initialize the instructions from a given backend.
+
+        Args:
+            backend: The backend from which to get the configuration.
+        """
+
+        self._n_qubits = backend.configuration().num_qubits
+        self._n_uchannels = backend.configuration().n_uchannels
+        self._config = backend.configuration()
+        self._params = {'qubit_freq': {}}
+        self._schedules = {}
+        self.backend = backend
+
+        # Populate the qubit frequency estimates
+        now = datetime.now()
+        for qubit, freq in enumerate(backend.defaults().qubit_freq_est):
+            val = ParameterValue(freq, now)
+            self.add_parameter_value('qubit_freq', val, DriveChannel(qubit))
+
+    def schedules(self) -> pd.DataFrame:
+        """
+        Return the schedules in self in a data frame to help
+        users manage their schedules.
+
+        Returns:
+            data: A pandas data frame with all the schedules in it.
+        """
+        data = []
+        for name, schedule in self._schedules.items():
+            data.append({'name': name,
+                         'schedule': schedule,
+                         'parameters': schedule.parameters})
+
+        return pd.DataFrame(data)
+
+    def parameters(self, names: Optional[List[str]] = None,
+                   chs: Optional[List[PulseChannel]] = None) -> pd.DataFrame:
+        """
+        Returns the parameters as a pandas data frame.
+        This function is here to help users manage their parameters.
+
+        Args:
+            names: The parameter names that should be included in the returned
+                table. If None is given then all names are included.
+            chs: The channels that should be included in the returned table.
+                If None is given then all channels are returned.
+
+        Returns:
+            data: A data frame of parameter values.
+        """
+
+        data = []
+
+        if names is None:
+            names = self._params.keys()
+
+        for name in names:
+            params_name = self._params.get(name, {})
+
+            if chs is None:
+                chs = params_name.keys()
+
+            for ch in chs:
+                for value in params_name.get(ch, {}):
+                    value_dict = dataclasses.asdict(value)
+                    value_dict['channel'] = ch.name
+                    value_dict['parameter'] = name
+
+                    data.append(value_dict)
+
+        return pd.DataFrame(data)
+
+    def add_schedules(self, schedules: Union[Schedule, List[Schedule]]):
```

Comment:
Why we need list input?

Reply:
It is not needed per say but it makes it easier to use if you have several schedule to add. I first started with schedule only but quickly found myself adding several schedules at a time so allowing list made things easier.